test/Microsoft.AspNetCore.Routing.Tests/Matchers/DispatcherAssert.cs

```diff
@@ -20,6 +20,11 @@ public static void AssertMatch(IEndpointFeature feature, Endpoint expected, Rout
                 throw new XunitException($"Was expected to match '{expected.DisplayName}' but did not match.");
             }
 
+            if (feature.Values == null)
+            {
+                throw new XunitException("Values is null.");
+            }
```

Comment:
This was missing.  A match should always produce route values, even if empty. It was unfair since some of the test implementations didn't do this.